### PR TITLE
Go Sample Initial Draft

### PR DESCRIPTION
**Pull request by sandeepsnairms**

### CODE REVIEW

File: Go/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Azure Management Go SDK Samples for Azure Cosmos DB
+
+Repository of Azure Management Go SDK samples for creating and updating Azure Cosmos DB resources.
+
+## Azure SDK for Python for Azure Cosmos DB
+
+You can find the source code and for this and other Azure SDKs at:
+[azure-mgmt-cosmosdb](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/cosmos/azure-mgmt-cosmosdb)
```
Comment (markjbrown) [inline Go/README.md]: can you change the package name and URL for the azure sdk for go repo?

File: Go/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+# Azure Management Go SDK Samples for Azure Cosmos DB
+
+Repository of Azure Management Go SDK samples for creating and updating Azure Cosmos DB resources.
+
+## Azure SDK for Python for Azure Cosmos DB
```
Comment (markjbrown) [inline Go/README.md]: change title to Go from Python

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Go/go.mod
```
Comment (markjbrown) [inline .gitignore]: can you create a .gitignore in the Go folder versus updating the one at root? I need to move that one to /CSharp as its for .NET

File: Go/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+
+## Prerequisites
+ - Go 1.16 or later
+
+## Setup
+
+### Windows
+
+1. Download the Go installer from the official website.
```
Comment (markjbrown) [inline Go/README.md]: add URL to the website

File: Go/README.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+go get github.com/spf13/viper
+go get github.com/google/uuid
+
+```
+
+The go get command downloads the specified packages and adds them to your go.mod file.
+
+### Configuration
+
+Update the `appsettings.json` in the project directory with the following content:
```
Comment (markjbrown) [inline Go/README.md:69]: include a local named version of this to the .gitignore so people don't accidentally check in secrets into their fork and update instructions to create a version of that file instead for config information.

File: Go/main.go [hunk 6]
```diff
@@ -0,0 +1,489 @@
...
+					IsZoneRedundant:  to.BoolPtr(false),
+				},
+			},
+			Capabilities: []*armcosmos.Capability{
+				{
+					Name: to.StringPtr("EnableNoSQLVectorSearch"),
+				},
+			},
+			DatabaseAccountOfferType: to.StringPtr("Standard"),
+			DisableLocalAuth:         to.BoolPtr(false),
```
Comment (markjbrown) [inline Go/main.go]: this should be true

File: Go/main.go [hunk 7]
```diff
@@ -0,0 +1,489 @@
...
+	databaseClient, err := armcosmos.NewSQLResourcesClient(subscriptionID, credential, nil)
+	if err != nil {
+		log.Fatalf("failed to create cosmos db database client: %v", err)
+	}
+
+	if _, err := databaseClient.GetSQLDatabase(ctx, resourceGroupName, accountName, databaseName, nil); err != nil {
+		log.Fatalf("failed to get cosmos db database: %v", err)
+	}
+
+	pollerResp, err := containerClient.BeginCreateUpdateSQLContainer(ctx, resourceGroupName, accountName, databaseName, containerName, properties, nil)
```
Comment (markjbrown) [inline Go/main.go:253]: how does this work? shouldn't this be databaseClient making this call? I don't see a reference to containerClient above being created.

Comment (sandeepsnairms) [reply Go/main.go:253]: containerClient, err := armcosmos.NewSQLResourcesClient(subscriptionID, credential, nil) is the first line in the method.

File: Go/main.go (hunk 6 above)
Comment (sandeepsnairms) [reply Go/main.go]: Yes, this was done for testing.

File: Go/README.md (hunk 4 above)
Comment (sandeepsnairms) [reply Go/README.md]: updated

File: Go/README.md (hunk 1 above)
Comment (sandeepsnairms) [reply Go/README.md]: Updated

File: Go/README.md (hunk 2 above)
Comment (sandeepsnairms) [reply Go/README.md]: Updated

File: Go/README.md (hunk 5 above)
Comment (sandeepsnairms) [reply Go/README.md:69]: Updated

File: .gitignore (hunk 3 above)
Comment (sandeepsnairms) [reply .gitignore]: Done

**Review by markjbrown — APPROVED** [review]
Looks awesome Sandeep! Thank you!!!

**Pull request merged**